ModelBase/src/persistence/NodeIdMap.cpp

```diff
@@ -28,10 +28,17 @@
 
 namespace Model {
 
-NodeIdType NodeIdMap::id(const Node* node)
+bool NodeIdMap::containsNode(const Node* node)
 {
 	QHash<const Node*, NodeIdType>::const_iterator iter{nodeToId.find(node)};
-	if ( iter != nodeToId.end() ) return *iter;
+	if ( iter != nodeToId.end() ) return true;
```

Comment:
Why not just replace the entire method body with one line:
`return nodeToId.contains(node);`

Since it's so short, you might as well inline this directly in the .h file.

In general, the mark of a good programmer is that they know their libraries well. I suppose Qt is new for you, so every time you want to use an unfamiliar Qt class, make sure to look up its APIs to know what's available.